checker-rs/src/notebook_client.rs

```diff
@@ -0,0 +1,351 @@
+        self.user.as_mut().unwrap().note_id = Some(note_id.to_string());
+
+        Ok(())
+    }
+
+    #[instrument("GET_NOTE")]
+    pub async fn get_note(&mut self) -> CheckerResult<String> {
+        info!(
+            "Retirieving Note for user {:?}",
+            self.user.as_ref().expect("Getting note of NONE-User")
+        );
+
+        let conn = &mut self.conn;
+        let note_id = self
+            .user
+            .as_ref()
+            .unwrap()
+            .note_id
+            .as_ref()
+            .expect("No NoteId for saved user");
+
+        let mut response_buf = Vec::with_capacity(256);
+        async {
+            let request = format!("get {}\n", note_id);
+            debug!("request: '{}'", request);
+            conn.write_all(request.as_bytes()).await?;
+            conn.flush().await?;
+
+            read_until_slice(conn, PROMPT, &mut response_buf).await
+        }
+        .await
+        .map_err(|e| {
+            info!("Socketerror when retrieving note: {:?}", e);
+            CheckerErrorUnfilled::from(e).with_message("Connection error upon fetching note")
+        })?;
+
+        response_buf.truncate(response_buf.len() - 3);
+        Ok(String::from_utf8(response_buf)?)
+    }
+
+    #[instrument("GET_HELP")]
+    pub async fn get_help(&mut self) -> CheckerResult<String> {
+        info!("Getting help");
+
+        let conn = &mut self.conn;
+        let mut response_buf = Vec::with_capacity(512);
+        async {
+            conn.write_all(b"help\n").await?;
+            conn.flush().await?;
+
+            read_until_slice(conn, PROMPT, &mut response_buf).await
+        }
+        .await
+        .map_err(|e| {
+            warn!("Socketerror upon getting help text {:?}", e);
+            CheckerErrorUnfilled::from(e).with_message("Connection error upon getting help")
+        })?;
+        response_buf.truncate(response_buf.len() - 3);
+        Ok(String::from_utf8(response_buf)?)
+    }
+
+    #[instrument("GET_USERS")]
+    pub async fn get_users(&mut self) -> CheckerResult<Vec<String>> {
+        info!("Listing users");
+
+        let conn = &mut self.conn;
+        let mut response_buf = Vec::with_capacity(512);
+        async {
+            conn.write_all(b"user\n").await?;
+            conn.flush().await?;
+
+            read_until_slice(conn, PROMPT, &mut response_buf).await
+        }
+        .await
+        .map_err(|e| {
+            warn!("Socketerror upon getting help text {:?}", e);
+            CheckerErrorUnfilled::from(e).with_message("Connection error upon getting help")
+        })?;
+
+        response_buf.truncate(response_buf.len() - 3);
+        let users = String::from_utf8(response_buf)?;
+
+        let user_arr: Vec<String> = users
+            .split('\n')
+            .filter_map(|line| {
+                line.split(": ")
+                    .nth(1)
+                    .map(std::string::ToString::to_string)
```

Comment:
(personal style:) would `use std::string::ToString` to make this a bit shorter

Reply:
that's what clippy recommended ^^

Reply:
Think clippy accepts both, it's just personal style to use the short version. Both ways are correct, oviously

Reply:
(feel free to resolve)

Reply:
310 |                     .map(std::string::to_string)
    |                                       ^^^^^^^^^ not found in `std::string`

Reply:
I actually meant `.map(ToString::to_string)`